src/api/gpu/ucx/nixl_device.cuh

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _NIXL_DEVICE_CUH
+#define _NIXL_DEVICE_CUH
+#include <nixl_types.h>
+
+struct nixlGpuXferStatusH;
+
+struct nixlGpuSignal {
+    uint64_t inc = 0;
+    uint64_t remote_addr = 0;
+};
+
+/**
+ * @enum  nixl_gpu_level_t
+ * @brief An enumeration of different levels for GPU transfer requests.
+ */
+enum class nixl_gpu_level_t : uint64_t {
+    NIXL_GPU_LEVEL_THREAD,
+    NIXL_GPU_LEVEL_WARP,
+    NIXL_GPU_LEVEL_BLOCK,
+    NIXL_GPU_LEVEL_GRID
```

Comment:
no need in `NIXL_GPU_LEVEL_` prefix with scoped enum class, because you are supposed to access it like `nixl_gpu_level_t::THREAD`